From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrei Borzenkov <arvidjaar@gmail.com>
Date: Fri, 19 Jun 2015 18:38:25 +0300
Subject: [PATCH] xfs: silence Coverity overflow warning

inode size cannot really overflow integer, but Coverity does not know it.
CID: 96602
---
 grub-core/fs/xfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/fs/xfs.c b/grub-core/fs/xfs.c
index f00e43e7dc3..72492915533 100644
--- a/grub-core/fs/xfs.c
+++ b/grub-core/fs/xfs.c
@@ -361,7 +361,7 @@ grub_xfs_inode_offset (struct grub_xfs_data *data,
 static inline grub_size_t
 grub_xfs_inode_size(struct grub_xfs_data *data)
 {
-  return 1 << data->sblock.log2_inode;
+  return (grub_size_t)1 << data->sblock.log2_inode;
 }
 
 /*
